clang/lib/Basic/Targets/PPC.cpp

```diff
@@ -402,9 +402,16 @@ void PPCTargetInfo::getTargetDefines(const LangOptions &Opts,
     Builder.defineMacro("__VEC__", "10206");
     Builder.defineMacro("__ALTIVEC__");
   }
-  if (HasSPE) {
+  if (HasSPE)
     Builder.defineMacro("__SPE__");
+  if (HasSPE || FloatABI == SoftFloat)
     Builder.defineMacro("__NO_FPRS__");
+  if (FloatABI == SoftFloat) {
+    Builder.defineMacro("_SOFT_FLOAT");
+    Builder.defineMacro("_SOFT_DOUBLE");
+  } else {
+    Builder.defineMacro("__RSQRTE__");
+    Builder.defineMacro("__RSQRTEF__");
```

Comment:
Sorry, I am not familiar with these macros. But I don't think GCC will always define `__RSQRTEF__` macro for all PowerPC versions. With -mcpu=power4, this macro is not defined, see https://godbolt.org/z/6W9f3adT1

Reply:
Something like what I just pushed?